app/src/main/java/com/example/firstapplication/Activity/CryptoHomepageActivity.java

```diff
@@ -0,0 +1,126 @@
+package com.example.firstapplication.Activity;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.content.ContextCompat;
+import androidx.viewpager.widget.ViewPager;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+
+import com.example.firstapplication.Adapter.MpagerAdapter;
+import com.example.firstapplication.R;
+
+public class CryptoHomepageActivity extends AppCompatActivity implements View.OnClickListener {
+    private ViewPager mPager;
+    private int layouts[] = {
+            R.layout.crpyto_firstslide,R.layout.crypto_secondslide,
+            R.layout.crypto_thirdslide};
+    private MpagerAdapter mpagerAdapter;
+
+    //sliding dots
+    private LinearLayout dots_layout;
+    private ImageView dots[];
+
+    //buttons
+    private Button buttonNext, buttonSkip;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_crypto_homepage);
+
+        mPager = (ViewPager) findViewById(R.id.viewPager);
+        mpagerAdapter = new MpagerAdapter(layouts, this);
+        mPager.setAdapter(mpagerAdapter);
+
+        dots_layout = (LinearLayout) findViewById(R.id.dotsLayout);
+        createDots(0);
+
+        buttonNext = (Button) findViewById(R.id.btnNext);
+        buttonSkip = (Button) findViewById(R.id.btnSkip);
+
+        buttonNext.setOnClickListener(this);
+        buttonSkip.setOnClickListener(this);
+
+        mPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
+            @Override
+            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
+
+            }
+
+            @Override
+            public void onPageSelected(int position) {
+                createDots(position);
+                if (position == layouts.length -1 ) {
+                    buttonNext.setText("Start");
```

Comment:
Avoid hard coded "Strings" move 'Start' and 'Next' strings into the strings.xml

Reply:
alright